init-kafka/src/main/java/io/strimzi/init/kafka/Main.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.init.kafka;
+
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Map;
+
+public class Main {
+
+    private static final Logger log = LoggerFactory.getLogger(Main.class);
+
+    private static final String RACK_FILE_PATH = "./rack/rack.id";
+
+    public static void main(String[] args) {
+
+        // passed by the cluster operator when creating the init container
+        String rackTopologyKey = System.getenv("RACK_TOPOLOGY_KEY");
+        log.info("RACK_TOPOLOGY_KEY = {}", rackTopologyKey);
+
+        // using Downward API for getting current node
+        String nodeName = System.getenv("NODE_NAME");
+        log.info("NODE_NAME = {}", nodeName);
+
+        KubernetesClient client = new DefaultKubernetesClient();
+
+        Map<String, String> nodeLabels = client.nodes().withName(nodeName).get().getMetadata().getLabels();
```

Comment:
What happens if we do not have the rights to get the nodes? Will this fail with an appropriate error? People will try this without setting up the permissions, so we need to be sure it properly fails.

Reply:
It fails as the cluster operator fails when you forget to create the sa with related rights.

Reply:
Well ... I expect cases when the SA exists (user can normally create it without cluster admin, so SA creation will pass). But the ClusterRole / binding will not be there because that requires more rights. I assume that unhandled exception should probably cause it to fail. But wanted to make sure.

Reply:
Yeah I tried with having the SA but without the role and it works exactly as you are describing.